cvat-ui/src/actions/annotation-actions.ts

```diff
@@ -1671,3 +1768,85 @@ export function switchNavigationBlocked(navigationBlocked: boolean): AnyAction {
         },
     };
 }
+
+export function deleteFrameAsync(): ThunkAction {
+    return async (dispatch: ActionCreator<Dispatch>): Promise<void> => {
+        const state: CombinedState = getStore().getState();
+        const {
+            annotation: {
+                player: {
+                    frame: {
+                        data: frameData,
+                    },
+                },
+                job: {
+                    instance: jobInstance,
+                },
+            },
+            settings: {
+                player: { showDeletedFrames },
+            },
+        } = state;
+
+        try {
+            dispatch({
+                type: AnnotationActionTypes.DELETE_FRAME,
+            });
+
+            await jobInstance.frames.delete(frameData.number);
+            const data = await jobInstance.frames.get(frameData.number);
```

Comment:
Maybe frames.delete should return a new instance?

Reply:
Getting a frame requires a little bit more arguments. I think this variant also looks normal.